File: resource/cpp/com/template/JiaSu.cpp
#include "JiaSu.h"
void JiaSu::read(ByteBuffer* buffer)
{
		this->type=buffer->readUTF();
		this->time=buffer->readUTF();
		short goods_count=buffer->readShort();
		if(goods_count>0)
		{
			for(int i=0;i<goods_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->goods.push_back(tmp);
			}
		}

}

File: resource/java/com/template/JiaSu.java
package com.template;
import java.io.DataInputStream;
public class JiaSu implements TemplateInterface
{
	public String type;
	public String time;
	public String[] goods;
	public void read(DataInputStream in)throws Exception
	{
		type=in.readUTF();
		time=in.readUTF();
		short goods_count=in.readShort();
		if(goods_count>0)
		{
			goods=new String[goods_count];
			for(int i=0;i<goods_count;i++)
			{
				goods[i]=in.readUTF();
			}
		}

	}
}
